index.js

```diff
@@ -278,8 +278,17 @@
          * Store Injection
          */
         function createStoreInjector(stores, component) {
+            
+            var statics = Object.getOwnPropertyNames(component).filter(function (propName) {
+				return typeof component[propName] === 'function';
```

Comment:
Good to know about the getOwnProperties issues.
On Jul 9, 2016 6:40 AM, "Prashant Tiwari" <notifications@github.com> wrote:

> In index.js
> <https://github.com/mobxjs/mobx-react/pull/72#discussion_r70166063>:
>
> > @@ -278,8 +278,17 @@
> >           * Store Injection
> >           */
> >          function createStoreInjector(stores, component) {
> > +
> > +            var statics = Object.getOwnPropertyNames(component).filter(function (propName) {
> > +				return typeof component[propName] === 'function';
>
> @mattruby <https://github.com/mattruby> Just wanted to give a heads up,
> this fails in Safari because Object.getOwnPropertyNames apparently
> doesn't respect strict mode and returns arguments and caller also as
> properties. It would perhaps have been better to use Object.keys here.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mobxjs/mobx-react/pull/72/files/660af27cfe1de04e369a6d90139d1ae8d961b6c3#r70166063>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAIrcsAKAaDbdkVuU9wkQIEKY_BR5LD1ks5qT4jAgaJpZM4JCizE>
> .
>